### ISSUE TITLE
DynSurv() - Linear prediction for new patients

### ISSUE THREAD

**Issue by PradeepVirdee** [graemeleehickey/joineRML#76]
Hi Graeme,

I've fit a joint model using mjoint() and stored the model under the object name "jointmodel". I've obtained dynamic predictions using dynSurv() on the model building sample but also a validation sample containing data from new patients. My aim now is to calculate the model's calibration slope in the validation sample (which involves fitting a Cox model with linear predictions as the only covariate). For the patients used to build the model, I can extract their linear predictions from jointmodel$sfit$linear.predictors. Can I extract the linear prediction for each new patient in the validation sample? If not, is there a way I can work this out? 

Pradeep

**Comment by graemeleehickey** [comment]
I'm not sure I quite follow. jointmodel$sfit is just the Cox model for the
separate time-to-event model ignoring the joint-model component, X(t). So
jointmodel$sfit$linear.predictors is equivalent to fitting a Cox model with
fixed effects and pulling out the linear discriminant term.

**Comment by PradeepVirdee** [comment]
Yes, I figured that out shortly after posting the question. Sorry, at the time I posted, I thought jointmodel$sfit was a Cox model that included the joint model component. Say I wanted linear predictions from the Cox model that does include the joint model component, is that possible? Sorry if this is a silly question.

Pradeep

Mr Pradeep Virdee
NIHR Doctoral Research Fellow; Medical Statistician
Centre for Statistics in Medicine (CSM),
Botnar Research Centre, NDORMS,
Nuffield Orthopaedic Centre,
University of Oxford,
Windmill Road,
Oxford,
OX3 7LD

Tel: +44 (0)1865 737950
Email: ***@***.***<https://owa.nexus.ox.ac.uk/owa/redir.aspx?C=M9QJvEjCUkuxZIviXHPm2T5wkQ4vR9IIT0wsw9HEKA5FGwCUhzgJWNNdd4XrD5EdWxtXidh_3Ts.&URL=mailto%3apradeep.virdee%40csm.ox.ac.uk>
Web: https://www.ndorms.ox.ac.uk/team/pradeep-virdee

**Comment by graemeleehickey** [comment]
Since the linear term in the Cox model is a function of a time, you would need to construct it from the coefficients + the time-varying component, m(t).
